src/samplers/support/hmc_core.jl

```diff
@@ -185,12 +185,12 @@ end
 
 function _hmc_step(θ::AbstractVector{<:Real},
                    lj::Real,
-                   lj_func::Function,
-                   grad_func::Function,
-                   H_func::Function,
+                   lj_func,
+                   grad_func,
+                   H_func,
                    τ::Int,
                    ϵ::Real,
-                   momentum_sampler::Function;
+                   momentum_sampler;
```

Comment:
Same here. I guess the compiler changes should not affect the `momentum_sampler`. Or what is the rational behind that?

Reply:
You are probably right.